py4DSTEM/process/diskdetection/diskdetection_cuda.py

```diff
@@ -124,43 +128,116 @@ def find_Bragg_disks_CUDA(
     if _qt_progress_bar is not None:
         from PyQt5.QtWidgets import QApplication
 
-    # Loop over all diffraction patterns
     t0 = time()
-    for (Rx, Ry) in tqdmnd(
-        datacube.R_Nx,
-        datacube.R_Ny,
-        desc="Finding Bragg Disks",
-        unit="DP",
-        unit_scale=True,
-    ):
-        if _qt_progress_bar is not None:
-            _qt_progress_bar.setValue(Rx * datacube.R_Ny + Ry + 1)
-            QApplication.processEvents()
-        DP = datacube.data[Rx, Ry, :, :]
-        _find_Bragg_disks_single_DP_FK_CUDA(
-            DP,
-            probe_kernel_FT,
-            corrPower=corrPower,
-            sigma=sigma,
-            edgeBoundary=edgeBoundary,
-            minRelativeIntensity=minRelativeIntensity,
-            minAbsoluteIntensity=minAbsoluteIntensity,
-            relativeToPeak=relativeToPeak,
-            minPeakSpacing=minPeakSpacing,
-            maxNumPeaks=maxNumPeaks,
-            subpixel=subpixel,
-            upsample_factor=upsample_factor,
-            filter_function=filter_function,
-            peaks=peaks.get_pointlist(Rx, Ry),
-            get_maximal_points=get_maximal_points,
-            blocks=blocks,
-            threads=threads,
-        )
+    if batching:
+        # compute the batch size based on available VRAM:
+        max_num_bytes = cp.cuda.Device().mem_info[0]
+        # use a fudge factor to leave room for the fourier transformed data
+        # I have set this at 10, which results in underutilization of 
+        # VRAM, because this yielded better performance in my testing
+        batch_size = max_num_bytes // (bytes_per_pattern * 10)
```

Comment:
what other values did you try for this? Did you try datasets with different types and/or sized diffraction patterns? I wonder if there's way to pick this value dynamically? But seems sensible for now.

Reply:
A value of 5 will cause to VRAM to be nearly completely full, which seems to either cause the garbage collector to get invoked more often or to limit the available working RAM for the FFT. So below ~5 it just won't run at all, at 5 it runs perhaps 5% slower than at 10. If you keep adjusting it up, at some point you won't be delivering enough work to the GPU to saturate it at each FFT step, but already the FFT isn't the bottleneck anyway. So this number isn't critical